### DISCUSSION TITLE
Use simple and little http client and server on Android.

### DISCUSSION THREAD

**Discussion by peerless2012** [jupnp/jupnp#254] [Ideas]
Jetty is too heavy for Android platform, and will make the apk size become more big.

So, is it possible to use OKHttp as http client and NanoHTTPD as web server?

**Comment by wborn** [comment]
Yes you can replace Jetty by using a custom configuration instead of `AndroidUpnpServiceConfiguration`.

**Reply by liuzho** [reply]
What do you think about adding an `OkHttpClientImpl` under the `org.jupnp.android` package? I wrote a kotlin version, if this is necessary, I can convert it to java and submit a PR.
This can be done on the Android without relying on `jetty-client`. `OkHttp` is more popular on the Android without having to introduce `jettry-client`

**Reply by strange-qwq** [reply]
I think this is great, please do this for sure

**Comment by peerless2012** [comment]
Add a offical support for android tiny impl?

**Comment by gufunhua** [comment]
good idea， use netty for server
